applications/awards/views.py

```diff
@@ -1,12 +1,14 @@
-from django.shortcuts import render
+from django.shortcuts import render ,redirect
 from .models import Award
 
-
 # Create your views here.
 def index(request):
     return render(request, "awards/index.html")
 
 
 def award(request, id):
-    award = Award.objects.get(award_id=id)
-    return render(request, "awards/award.html", {"award": award})
+    try:
+        award = Award.objects.get(award_id=id)
+        return render(request, "awards/award.html", {"award": award})
```

Comment:
I think here the error is coming from the `render` function call when trying to render the page with a non-existent award. But we shouldn't try to render the page in that case in the first place.

So, instead of relying on the `render` call to throw an error, we should just check if `award` in an empty QuerySet, and if it is, redirect to the index page.

Reply:
Is this OK ?
     `try:`
        `award=get_object_or_404(Award , award_id=id)`
    `except:`
       ` return redirect('awards:index')`
    `return render(request, "awards/award.html", {"award": award})`

Reply:
Ohh, right! I completely forgot that the `.get()` method throws a `DoesNotExist` error if no matching data is found. In that case, your current implementation seems good.